helpers/utils.py

```diff
@@ -107,4 +119,58 @@ def UmaskNamedTemporaryFile(*args, **kargs):
     umask = os.umask(0o666)
     os.umask(umask)
     os.chmod(fdesc.name, 0o666 & ~umask)
-    return fdesc
\ No newline at end of file
+    return fdesc
+
+def setup_kubernetes_secret(properties_file: str, service_account_name: str) -> None:
+    if not properties_file:
+        return
+    kubeconfig = get_kube_config()
+    kubectl_cmd = get_kubectl_cmd()
+    secret_name = f"spark-client-sa-conf-{service_account_name or 'spark'}"
+    cmd = f"{kubectl_cmd} --kubeconfig {kubeconfig} create secret generic {secret_name} --from-env-file={properties_file}"
+    try:
+        subprocess.check_output(cmd, shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(
+            f'Cannot create secret. Probably bad or missing kubeconfig file {kubeconfig}')
+        logging.warning(
+            f'Please set the KUBECONFIG environment variable to point to the kubeconfig. Or fix the default kubeconfig {kubeconfig}')
+        logging.error(e.output)
+        sys.exit(EXIT_CODE_BAD_KUBECONFIG)
+
+def retrieve_kubernetes_secret(service_account_name: str, keys: List[str]) -> None:
+    secret_name = f"spark-client-sa-conf-{service_account_name or 'spark'}"
+    kubeconfig = get_kube_config()
+    kubectl_cmd = get_kubectl_cmd()
+
+    if not keys or len(keys) == 0:
```

Comment:
I would break the business logic into parsing the properties AND printing the values. 

Note that printing the values could be common to any property information (provided usually as dict), whereas reasing properties from k8s is generally what we would like to do when generating the dynamic configuration